hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -110,44 +106,52 @@ case class EmitRegion(mb: EmitMethodBuilder, region: Code[Region]) {
   def baseRegion: Code[Region] = mb.getArg[Region](1)
 }
 
-case class EmitTriplet(setup: Code[Unit], m: Code[Boolean], v: Code[_]) {
+case class EmitValue(m: Code[Boolean], pv: PValue) {
+  def v: Code[_] = pv.code
+
+  def value[T]: Code[T] = coerce[T](v)
+}
+
+case class EmitTriplet(setup: Code[Unit], m: Code[Boolean], pv: PValue) {
+  def v: Code[_] = pv.code
+
   def value[T]: Code[T] = coerce[T](v)
 }
 
 case class EmitArrayTriplet(setup: Code[Unit], m: Option[Code[Boolean]], addElements: Code[Unit])
 
 case class ArrayIteratorTriplet(calcLength: Code[Unit], length: Option[Code[Int]], arrayEmitter: Emit.F => EmitArrayTriplet) {
   def arrayEmitterFromBuilder(sab: StagedArrayBuilder): EmitArrayTriplet = {
-    arrayEmitter( { (m: Code[Boolean], v: Code[_]) => m.mux(sab.addMissing(), sab.add(v)) } )
+    arrayEmitter( { (m: Code[Boolean], v: PValue) => m.mux(sab.addMissing(), sab.add(v.code)) } )
   }
 
   def toEmitTriplet(mb: MethodBuilder, aTyp: PArray): EmitTriplet = {
     val srvb = new StagedRegionValueBuilder(mb, aTyp)
 
     length match {
       case Some(len) =>
-        val cont = { (m: Code[Boolean], v: Code[_]) =>
+        val cont = { (m: Code[Boolean], v: PValue) =>
           coerce[Unit](
             Code(
               m.mux(
                 srvb.setMissing(),
-                srvb.addIRIntermediate(aTyp.elementType)(v)),
+                srvb.addIRIntermediate(aTyp.elementType)(v.code)),
```

Comment:
pt is a PhysicalValue.  v is a Code[_].  They aren't interchangeable.

Right now PhysicalValue is a wrapper for Code[_], because that's how values are represented currently.  This code will have to change before we can support more general pvalue representations.

Reply:
Oh, v.pt isn’t the PType corresponding to v.code?

Reply:
Sorry, I misunderstood what you were asking for.  Yes, pt is the PType.  I made the change.